Library/Homebrew/test/messages_spec.rb

```diff
@@ -2,49 +2,95 @@
 require "spec_helper"
 
 describe Messages do
-  before do
-    @m = Messages.new
-    f_foo = formula("foo") do
-      url "https://example.com/foo-0.1.tgz"
-    end
-    f_bar = formula("bar") do
-      url "https://example.com/bar-0.1.tgz"
-    end
-    f_baz = formula("baz") do
-      url "https://example.com/baz-0.1.tgz"
-    end
-    @m.formula_installed(f_foo, 1.1)
-    @m.record_caveats(f_foo, "Zsh completions were installed")
-    @m.formula_installed(f_bar, 2.2)
-    @m.record_caveats(f_bar, "Keg-only formula")
-    @m.formula_installed(f_baz, 3.3)
-    @m.record_caveats(f_baz, "A valid GOPATH is required to use the go command")
-  end
+  let(:messages) { Messages.new }
+  let(:test_formula) { formula("foo") { url("https://example.com/foo-0.1.tgz") } }
 
-  it "has the right installed-formula count" do
-    expect(@m.formula_count).to equal(3)
+  describe "#record_caveats" do
+    it "adds a caveat" do
+      expect(messages.caveats).to be_empty
+      messages.record_caveats(test_formula, "Zsh completions were installed")
+      expect(messages.caveats).to_not be_empty
+    end
   end
 
-  it "has recorded caveats" do
-    expect(@m.caveats).to_not be_empty
-  end
+  describe "#formula_installed" do
+    it "counts the number of formulas installed" do
+      expect(messages.formula_count).to eq 0
+      messages.formula_installed(test_formula, 1.1)
+      expect(messages.formula_count).to eq 1
+    end
 
-  it "maintained the order of recorded caveats" do
-    caveats_formula_order = @m.caveats.map { |x| x[:formula] }
-    expect(caveats_formula_order).to eq(["foo", "bar", "baz"])
+    it "adds to install_times" do
+      expect(messages.install_times).to be_empty
+      messages.formula_installed(test_formula, 1.1)
+      expect(messages.install_times).to_not be_empty
+    end
   end
 
-  it "has recorded installation times" do
-    expect(@m.install_times).to_not be_empty
+  describe "#display_messages" do
+    it "calls #display_caveats" do
+      expect(messages).to receive(:display_caveats)
+      messages.display_messages
```

Comment:
This is not a meaningful test because it only tests internal logic.

Reply:
That's a good point! I will make changes to test the actual input from `#display_messages`. This makes me think that `#display_caveats` and `#display_install_times` should be marked as private. Do you agree?